app/controllers/referrals_controller.rb

```diff
@@ -18,15 +25,23 @@ def reject
   end
 
   def index
+    # getting all referrals that the current_user has been recommended (to_user)
+    @to_referrals = current_user.to_referrals.where(tied: nil).where(reject: nil).where(requested: nil)
 
-    # getting all referrals for the to_user where reject = nil
-    @to_referrals = current_user.to_referrals.where(reject: nil)
+    # Get all referrals where the current user has been requested (recommended_user)
+    @requested_referrals = current_user.recommended.where(tied: nil).where(reject: false).where(requested: true)
 
-    # mapping users recommended users to an array
-    @referrals = @to_referrals.map do |referral|
-      User.find_by id: referral.recommended_user_id
+    @ties = Tie.where('user1_id= :user OR user2_id= :user', {user: current_user})
```

Comment:
line 34 to 43 - I would put this in a method (called my_ties() for example) so that others could use it too. Like current_user.**my_ties()**

Reply:
I meant create a method in User model

Reply:
Something like this:
```
def my_tie_users

   @user_ids =  (Tie.where(user1: @u).pluck(:user2_id) + 
                 Tie.where(user2: @u).pluck(:user1_id) ).uniq
                  
   @users = @user_ids.map { |user_id| Users.find(user_id) }
   
   return @users
end
```